homeassistant/components/tplink/light.py

```diff
@@ -212,6 +228,7 @@ def get_features(self):
         """Determine all supported features in one go."""
         self._sysinfo = self.smartbulb.sys_info
         self._supported_features = 0
+        self._mac = self._sysinfo.get("mac") or self._sysinfo.get("mic_mac")
```

Comment:
This should be done in the backend library.

Also, could you please create an issue on the core, assuming that there was no error message indicating that the problem these not showing in the UI was because of a missing/invalid unique id? Such a warning/error should definitely be added as that mistake can be way too easy to make...

Reply:
This is handled by pyHS100 if you use the SmartDevice.mac property. Using these properties results in additional http calls to the device. Since this data is only pulled once per device, I've updated the code to use smart device properties for mac, alias and model to avoid future problems.

Reply:
Yes, I'm planning to fix that after https://github.com/GadgetReactor/pyHS100/pull/151 gets merged, but I haven't found time to push that through. After that gets merged, I'll change the behavior of the library to cache the responses from the device where appropriate, the tests I created are there in place to make sure nothing breaks (besides the API in some cases).